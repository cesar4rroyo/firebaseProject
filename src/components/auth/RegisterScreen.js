import React from "react";
import { Link } from "react-router-dom";

export const RegisterScreen = () => {
  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form>
        <input
          type="text"
          name="name"
          id="name"
          placeholder="Name"
          className="auth__input"
          autoComplete="off"
        />
        <input
          type="text"
          name="email"
          id="email"
          placeholder="Email"
          className="auth__input"
          autoComplete="off"
        />
        <input
          type="password"
          name="password"
          id="password"
          placeholder="Password"
          className="auth__input"
        />
        <input
          type="password"
          name="passwprd2"
          id="passwprd2"
          placeholder="Confirm Password"
          className="auth__input"
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>
        <Link to="/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};
