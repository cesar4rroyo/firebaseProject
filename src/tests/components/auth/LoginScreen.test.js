import React from 'react';
import { LoginScreen } from '../../../components/auth/LoginScreen';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import thunk from "redux-thunk"
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import { startGoogleLogin, startLoginEmailPassword } from '../../../actions/auth';

jest.mock("../../../actions/auth", () => ({
    startGoogleLogin: jest.fn(),
    startLoginEmailPassword: jest.fn()
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null

    }
}

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <LoginScreen />
        </MemoryRouter>
    </Provider>

);

describe('Test in <LoginScreen />', () => {
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    })

    test('should do done correctly ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should de disparar la accion startLogingSCreen', () => {
        wrapper.find(".google-btn").prop("onClick")();
        expect(startGoogleLogin).toHaveBeenCalled();
    })

    test('debe de disparar el startLogin con sus argumentos respectivos', () => {
        wrapper.find("form").prop("onSubmit")({
            preventDefault() { }
        });
        expect(startLoginEmailPassword).toHaveBeenCalledWith("cesar@gmail.com", "123456");
    })





})
