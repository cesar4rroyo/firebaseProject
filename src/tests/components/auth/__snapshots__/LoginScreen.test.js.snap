// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen /> should do done correctly  1`] = `
Array [
  <h3
    className="auth__title"
  >
    Login
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="cesar@gmail.com"
    />
    <input
      className="auth__input"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      type="submit"
    >
      Login
    </button>
    <div
      className="auth__social-networks"
    >
      <p>
        Login whit social networks
      </p>
      <div
        className="google-btn"
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="google button"
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Sign in with google
          </b>
        </p>
      </div>
    </div>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Create new account
    </a>
  </form>,
]
`;
